nomad/structs/structs.go

```diff
@@ -2592,6 +2603,54 @@ func NewRestartPolicy(jobType string) *RestartPolicy {
 	return nil
 }
 
+const ReschedulePolicyMinInterval = 15 * time.Second
+
+// ReschedulePolicy configures how Tasks are rescheduled  when they crash or fail.
+type ReschedulePolicy struct {
+	// Attempts limits the number of rescheduling attempts that can occur in an interval.
+	Attempts int
+
+	// Interval is a duration in which we can limit the number of reschedule attempts.
+	Interval time.Duration
+
+	//TODO delay
+}
+
+func (r *ReschedulePolicy) Copy() *ReschedulePolicy {
+	if r == nil {
+		return nil
+	}
+	nrp := new(ReschedulePolicy)
+	*nrp = *r
+	return nrp
+}
+
+func (r *ReschedulePolicy) Validate() error {
+	var mErr multierror.Error
+	// Check for ambiguous/confusing settings
+	if r.Attempts < 0 {
+		multierror.Append(&mErr, fmt.Errorf("Attempts must be >= 0 (got %v)", r.Attempts))
+	}
+
+	if r.Interval.Nanoseconds() < ReschedulePolicyMinInterval.Nanoseconds() {
```

Comment:
I feel like you should guard this with:

```
if r.Attempts > 0  && <the rest> {
```

Because otherwise the UX for disabling the rescheduler is bad